src/coprocessor/codec/mysql/time/mod.rs

```diff
@@ -265,35 +263,60 @@ impl Time {
         self.time = time
     }
 
-    fn to_numeric_str(&self) -> String {
+    /// Converts a `DateTime` to printable string representation
+    #[inline]
+    pub fn to_numeric_string(&self) -> String {
         if self.time_type == TimeType::Date {
-            // TODO: pure calculation should be enough.
-            format!("{}", self.time.format("%Y%m%d"))
+            if self.is_zero() {
```

Comment:
hmm. Why not simply use date format functions as TiDB does?

Reply:
The result of `zero` is different in TiDB/TiKV. 😿